packages/typegpu/src/std/numeric.ts

```diff
@@ -804,31 +806,41 @@ export const normalize = createDualImpl(
   'normalize',
 );
 
-type PowOverload = {
-  (base: number, exponent: number): number;
-  <T extends AnyFloatVecInstance>(base: T, exponent: T): T;
-};
-
-export const pow: PowOverload = createDualImpl(
-  // CPU implementation
-  <T extends AnyFloatVecInstance | number>(base: T, exponent: T): T => {
-    if (typeof base === 'number' && typeof exponent === 'number') {
-      return (base ** exponent) as T;
-    }
-    if (
-      typeof base === 'object' &&
-      typeof exponent === 'object' &&
-      'kind' in base &&
-      'kind' in exponent
-    ) {
-      return VectorOps.pow[base.kind](base, exponent) as T;
-    }
-    throw new Error('Invalid arguments to pow()');
+function powCpu(base: number, exponent: number): number;
+function powCpu<T extends AnyFloatVecInstance>(
+  base: T,
+  exponent: T,
+): T;
+function powCpu<T extends AnyFloatVecInstance | number>(
+  base: T,
+  exponent: T,
+): T {
+  if (typeof base === 'number' && typeof exponent === 'number') {
+    return (base ** exponent) as T;
+  }
+  if (
+    typeof base === 'object' &&
+    typeof exponent === 'object' &&
+    'kind' in base &&
+    'kind' in exponent
+  ) {
```

Comment:
Tiny nit, more resilient to refactors
```suggestion
  if (isVecInstance(base) && isVecInstance(exponent)) {
```